.travis/shellchecker

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Wrapper for running `shellcheck` over projects.
+
+shopt -s extglob
+cd "$(dirname "$0")/../" || exit 1
+
+# find shell files under the specified directory
+find_files () {
+    FPATH="$1"
+    grep --color=never -rl '^#\!.*bash' "${FPATH}" | sed 's/^\.\///' &
+    find "${FPATH}" -name '*.sh' | sed 's/^\.\///'
+}
+
+main () {
+    # parse CLI
+    includes=()
+    excludes=()
+    args=()
+    while [[ "$#" -gt 0 ]]; do
+        if [[ ${1} == '--' ]]; then
+            shift
+            break
+        elif [[ ${1} == "--exclude" ]]; then
+            shift
+            excludes+=("$1")
+        elif [[ ${1} == "-*" ]]; then
+            args+=("$1")
+        else
+            mapfile -t files < <(find_files "${1}")
+            includes=("${includes[@]}" "${files[@]}")
+        fi
+        shift
+    done
+
+    # extract files list
+    files=()
+    for include in "${includes[@]}"; do
+        skip=0
+        for exclude in "${excludes[@]}"; do
+            if [[ "${include}" == "${exclude}"* ]]; then
+                skip=1
+                break
+            fi
+        done
+        if [[ $skip == 0 ]]; then
+            files+=("${include}")
+        fi
+    done
+
+    # sort $files removing duplicates
+    readarray -t files < <(for x in "${files[@]}"; do echo "$x"; done | sort -u)
+
+    # run shellcheck
+    shellcheck "$@" "${files[@]}"
+}
+
+# default configuration for linting cylc
+default () {
+    # run a strict check on all "functional" scripts
+    main '.' \
+        --exclude 'tests/' \
+        --exclude 'lib/parsec/tests' \
+        --exclude 'lib/cylc/job.sh' \
+        --exclude 'etc/dev-bin/live-graph-movie.sh' \
+        -- -e SC1090
+    main 'lib/cylc/job.sh' -- -e SC1090 -s bash
+    main 'lib/cylc/job.sh' -- -e SC1090 -s ksh
```

Comment:
Do we want to remove [the deprecated] ksh support from the job file in Cylc8?

Reply:
I'd remove it.  I think we kept it in due to rampant use of ksh at your site, no?

Reply:
I don't think it was our site.

Reply:
Definitely not us. All platforms on our site support Bash. We recommend our users to only write Bash scripts in Cylc's `suite.rc` since the beginning.

Reply:
I've spun this off into #3071

Reply:
OK, that comment was probably a hang-over from my days of real UM wrangling, pre-Cylc - I remember a lot of MO ksh scripts (e.g. SCS!). Maybe we kept ksh in for @trwhitcomb (any comment Tim W?)

Reply:
@hjoliver It was us - given that version 7 went to bash-only for the actual script, we've been able to keep that while maintaining the korn shell scripts outside of Cylc.  OK by me to remove the ksh support internal to Cylc.

Reply:
I'll put up a new PR for this shortly...